src/actions/mod.rs

```diff
@@ -180,32 +180,69 @@ impl InitActionContext {
         let line = self.vfs.load_line(&file_path, pos.row).unwrap();
         trace!("line: `{}`", line);
 
-        let start_pos = {
-            let mut col = 0;
-            for (i, c) in line.chars().enumerate() {
-                if !(c.is_alphanumeric() || c == '_') {
-                    col = i + 1;
-                }
-                if i == pos.col.0 as usize {
-                    break;
-                }
-            }
-            trace!("start: {}", col);
-            span::Position::new(pos.row, span::Column::new_zero_indexed(col as u32))
-        };
+        let (start, end) = find_word_at_pos(&line, &pos.col);
+        trace!("start: {}, end: {}", start.0, end.0);
 
-        let end_pos = {
-            let mut col = pos.col.0 as usize;
-            for c in line.chars().skip(col) {
-                if !(c.is_alphanumeric() || c == '_') {
-                    break;
-                }
-                col += 1;
-            }
-            trace!("end: {}", col);
-            span::Position::new(pos.row, span::Column::new_zero_indexed(col as u32))
-        };
-
-        Span::from_positions(start_pos, end_pos, file_path)
+        Span::from_positions(span::Position::new(pos.row, start),
+                             span::Position::new(pos.row, end),
+                             file_path)
     }
 }
+
+/// Represents a text cursor between characters, pointing at the next character
+/// in the buffer.
+type Column = span::Column<span::ZeroIndexed>;
+/// Represents a `(start, end)` range between `start` and `end` cursors.
+/// For example (4, 4) means an empty selection starting after first 4 characters.
+type Range = (Column, Column);
+/// Returns a `Range` for a found word inside `line` at which `pos` text cursor
```

Comment:
Is the function comment too much? I tried to document intermediate types to better explain idea behind them, but this left me with only an obvious, I think, comment for the function.

Reply:
I would not define `Range` here - rls-span already defines a type called `Range` and it is quite different from this one, so I think it would be confusing. I would continue to return `(Column, Column)`, but add a comment as to what that means (the comments are fine, would just need to move them)